packages/kbn-health-gateway-server/src/kibana/routes/root.ts

```diff
@@ -149,13 +159,11 @@ function configureFetch(kibanaConfig: KibanaConfig) {
       signal: controller.signal,
       redirect: 'manual',
     };
+
     try {
-      const response = await nodeFetch(url, fetchOptions);
-      clearTimeout(timeoutId);
-      return response;
-    } catch (e) {
+      return await nodeFetch(url, fetchOptions);
+    } finally {
```

Comment:
Would be useful to see the end-to-end behaviour once we have integration tests but can you explain why you removed the `throw e`?

I think if `nodeFetch` rejects because of e.g. a connection error we'll end up in the `finally` clause but won't return the rejected promise so the error will just get silently lost.

Reply:
Omitting the `catch` block here doesn't suppress the exception. It will remove an unnecessary entry from the stack trace. We have an `await` statement that will reject the returned promise.

Reply:
```
async function testAwaitReject() {
  try {
    return await Promise.reject('i reject you');
  } catch (e) {
    console.log('we caught', e);
  }
}

async function run() {
  const result = await testAwaitReject();
  console.log('result was', result);
}

run();
```

Prints:
> we caught i reject you
> result was undefined

So I think this means that fetching with `await configureFetch()()` wouldn't reject but just return undefined.

Reply:
I missed that the `finally` won't handle the exception so it will just bubble up :+1: